src/GitVersionCore/SemanticVersionFormatValues.cs

```diff
@@ -0,0 +1,116 @@
+﻿namespace GitVersion
+{
+    public class SemanticVersionFormatValues
```

Comment:
We need this class in order to generate the VersionVariables class since the types we're using int he formatting need to come from a constructed object.  And if we exposed the VersionVariables object as what's valid for formatting, then someone could do {InformationalVersion} and get themselves into an infinite loop.  So this gives us type safety of what's allowed to be formatted as well as getting the formatting done to construct the VersionVariables type.  I agree it's redundant on most fields (since VersionVariables now just becomes a passthrough/holder for most of the values calculated in the SemanticVersionFormatValues class.  Open to suggestions to minimize here if you have them.